### PR TITLE
os/net: some modifications in common for stdk

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
os/arch/xtensa: init pipe during booting for xtensa
os/net: Add LWIP_NETIF_TX_SINGLE_PBUF in lwipopts for stdk
external/mbedtls: Enable some mbedtls config for stdk
external/json: Add GetStringValue API for cJSON

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [fae71f4271f8509dcd3a2719cee72613d50e0346] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fae71f4271f8509dcd3a2719cee72613d50e0346] - Code Rule Check OK.

File: external/include/mbedtls/config.h [hunk 1]
```diff
@@ -2866,7 +2866,9 @@
 #define MBEDTLS_LIGHT_DEVICE
 #if defined(MBEDTLS_LIGHT_DEVICE)
 
+#if !defined(CONFIG_STDK_IOT_CORE)
```
Comment (SeonghoByeon) [inline external/include/mbedtls/config.h]: In my opinion, STDK related configuration should be re-written in another file, such as config_stdk.h, and this header can be mapped into MBEDTLS_CONFIG_FILE.

File: os/include/net/lwip/lwipopts.h [hunk 2]
```diff
@@ -527,6 +527,19 @@
 #define MEMP_SEPARATE_POOLS	CONFIG_NET_MEMP_SEPARATE_POOLS
 #endif
 
+#if defined(CONFIG_STDK_IOT_CORE)
```
Comment (SeonghoByeon) [inline os/include/net/lwip/lwipopts.h]: lwipopt.h should be related only with the LwIP's own features, not STDK.

**Comment by seinfra** [comment]
Target : [2090ab5ab5b0f45008d0c2c1f9273949188aedd9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2090ab5ab5b0f45008d0c2c1f9273949188aedd9] - Code Rule Check (C++) OK.

File: external/include/mbedtls/config.h (hunk 1 above)
Comment (SRCN-LiQin) [reply external/include/mbedtls/config.h]: I will refine this config without changing this file, and this file modification will be removed, thank you

File: os/include/net/lwip/lwipopts.h (hunk 2 above)
Comment (SRCN-LiQin) [reply os/include/net/lwip/lwipopts.h]: new Kconfig item NET_LWIP_SINGLE_PBUF is added for this, please check the modification and share your opinion, thanks

Comment (SeonghoByeon) [reply os/include/net/lwip/lwipopts.h]: Got it.

**Pull request merged**